docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+
+- spec.url: Stores the url of the remote cluster. The field is only required for the source cluster object.
+
+## MigHooks
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigHook
+metadata:
+  generateName: test-
+  name: test-btmv5
+  namespace: openshift-migration
+spec:
+  custom: false
+  image: 'quay.io/konveyor/hook-runner:latest'
+  playbook: >-
+    LSBuYW1lOiBTbGVlcCBmb3IgMTAgc2Vjb25kcyBhbmQgY29udGludWUgd2l0aCBwbGF5CiAgd2FpdF9mb3I6CiAgIadsfwerq0OiAxMA==
+  targetCluster: source
+```
+
+- spec.activeDeadlineSeconds: Specifies the highest amount of time for which the hook will run.
+
+- spec.custom: Specifies whether the hook is a custom Ansible playbook or a pre-built image. This is a required field.
+
+- spec.image: Specifies the image of the hook to be executed. This is a required field.
+
+- spec.playbook: Specifies the contents of the custom Ansible playbook in base64 format, it is used in conjunction with the custom boolean flag.
+
+- spec.targetCluster: Specifies the cluster on which the hook is to be executed. This is a required field.
+
+
+
+## MigMigrations
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigMigration
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migmigration-sample
+  namespace: openshift-migration
+spec:
+  # [!] Set 'canceled: true' will cancel the migration
+  canceled: false
+  # [!] Set 'rollback: true' will rollback the migration's plan
+  rollback: true
+  # [!] Set 'stage: true' to run a 'Stage Migration' and skip quiescing of Pods on the source cluster.
+  stage: false
+  # [!] Set 'quiescePods: true' to scale down Pods on the source cluster after the 'Backup' stage of a migration has finished
+  quiescePods: false
+  # [!] Set 'keepAnnotations: true' to retain labels and annotations applied by the migration
+  keepAnnotations: false
+
+  migPlanRef:
+    name: migplan-sample
+    namespace: openshift-migration
+```
+
+- spec.canceled: Invokes the cancel migration operation, when set to true the migration controller switches to cancel itinerary. This field can be used on-demand to cancel the running migration.
+
+- spec.keepAnnotations: Specifies whether to retain the annotations set by the migration controller or not.
+
+- spec.migPlanRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.quiescePods: Specifies whether to quiesce the application pods before migration or not.
+
+- spec.rollback: Invokes the rollback migration operation, when set to true the migration controller switches to rollback itinerary. This field needs to be set prior to creation of a MigMigration.
+
+- spec.stage: Invokes the stage operation, when set to true the migration controller switches to stage itinerary. This is a required field.
+
+- spec.verify: Specifies whether to verify the health of the migrated pods or not.
```

Comment:
@eriknelson @shubham-pampattiwar 

spec.verify: What are the implications of this parameter? Will it reduce performance? What aspect of the migrated pods is checked? Does it return just the status of the pods or more information?

Reply:
@apinnick  It's' done to validate the pods are in a healthy state. It does add extra validation so will impact performance. The intention is to check if pods are in a Running state. It does return the pod names if they are unhealthy